symforce/slam/imu_preintegration/manifold_symbolic.py

```diff
@@ -0,0 +1,346 @@
+        (DR_D_gyro_bias * delta_gyro_bias).to_storage(), epsilon
+    )
+    corrected_Dv = Dv + Dv_D_gyro_bias * delta_gyro_bias + Dv_D_accel_bias * delta_accel_bias
+    corrected_Dp = Dp + Dp_D_gyro_bias * delta_gyro_bias + Dp_D_accel_bias * delta_accel_bias
+
+    res_R = (corrected_DR.inverse() * pose_i.R.inverse() * pose_j.R).to_tangent(epsilon)
+    res_v = pose_i.R.inverse() * (vel_j - vel_i - gravity * dt) - corrected_Dv
+    res_p = (
+        pose_i.R.inverse() * (pose_j.t - pose_i.t - vel_i * dt - sf.S(1) / 2 * gravity * dt ** 2)
+        - corrected_Dp
+    )
+
+    res = sf.V9(*res_R, *res_v, *res_p)
+
+    def upper_triangular_mul(ut: sf.M99, v: sf.V9) -> sf.V9:
+        """
+        Returns ut * v. Use to avoid expressions containing the zero-valued lower triangular
+        portion of ut.
+
+        Precondition: ut is upper triangular
+        """
+        return sf.V9([(ut[k, k:] * v[k:, 0])[0] for k in range(9)])
+
+    return upper_triangular_mul(sqrt_info, res)
+
+
+def right_jacobian(tangent: sf.V3, epsilon: T.Scalar) -> sf.M33:
+    """
+    The right jacobian J(v) is d/du Log(Exp(v)^{-1} Exp(v + u)), i.e., a matrix such that
+    Exp(v + dv) ~= Exp(v) Exp(J(v) dv), where v and u are tangent vectors of SO(3).
+
+    Returns J(tangent).
+    """
+    norm = tangent.norm(epsilon)
+    tangent_hat = sf.Matrix.skew_symmetric(tangent)
+    out = sf.M33.eye()
+    out += ((1 - sf.cos(norm)) / (norm ** 2)) * tangent_hat
+    out += ((norm - sf.sin(norm)) / (norm ** 3)) * (tangent_hat * tangent_hat)
+    return out
+
+
+def handwritten_state_D_old_state_gyro_accel(
+    DR: sf.Rot3, corrected_gyro: sf.V3, corrected_accel: sf.V3, dt: T.Scalar, epsilon: T.Scalar
+) -> T.Tuple[sf.Matrix, sf.Matrix, sf.Matrix]:
+    """
+    Calculates the derivatives of the new state (meaning the DR, Dv, and Dp) w.r.t. the previous state,
+    gyroscope measurement, and the accelerometer mesaurement.
+
+
+    Args:
+        DR (sf.Rot3): Preintegrated DR of the previous state
+        corrected_gyro (sf.V3): gyroscope measurment corrected for IMU bias
+        corrected_accel (sf.V3): accelerometer measurement corrected for IMU bias
+        dt (T.Scalar): Time difference between the previous time step and the new time step
+        epsilon (T.Scalar): epsilon for numerical stability
+
+    Returns:
+        T.Tuple[sf.M99, sf.M93, sf.M93]: new_state_D_old_state, new_state_D_gyro_measurement, new_state_D_accel_measurement
+    """
+    # NOTE(Brad): I put each of the 3 derivatives in one function to make it easier to keep
+    # subexpressions similar to improve CSE
+
+    # calculate new_state_D_old_state
+    DR_update = sf.Rot3.from_tangent(corrected_gyro * dt, epsilon)
+    new_DR_D_old_DR = DR_update.inverse().to_rotation_matrix()
+    new_DR_D_old_Dv = sf.M33.zero()
+    new_DR_D_old_Dp = sf.M33.zero()
+
+    new_Dv_D_old_DR = -(DR.to_rotation_matrix() * sf.Matrix.skew_symmetric(corrected_accel * dt))
```

Comment:
Are the parens here important?  Ditto on line 151 below

Reply:
Nope. The only thing that's important is that this line and 151 are consistent. So I'll remove the parens from both (though, there is another set of parents on line 151 that look like (A * B) * dt / 2 which don't matter because that's what `A * B * dt / 2` evaluates to, but it is important to do that rather than `A * (B * dt / 2)`.

Reply:
Yep for `(A * B) * dt / 2` agree it makes sense to keep them to be explicit